File: src/main/java/Movie.java
public class Movie {
    public static final int REGULAR = 0;
    public static final int NEW_RELEASE = 1;
    public static final int CHILDRENS = 2;

    private String title;
    Price price;

    public Movie(String title, int priceCode) {
        this.title = title;
        this.setPriceCode(priceCode);
    }

    public int getPriceCode() {
        return price.getPriceCode();
    }

    public void setPriceCode(int priceCode) {
        switch (priceCode) {
            case REGULAR: price = new RegularPrice(); break;
            case NEW_RELEASE: price = new NewReleasePrice(); break;
            case CHILDRENS: price = new ChildrensPrice(); break;
        }
    }

    public String getTitle() {
        return title;
    }

}

File: src/main/kotlin/Movie.kt
import java.lang.IllegalArgumentException

class Movie(val title: String, priceCode: Int) {
    private lateinit var price: Price

    init {
        setPriceCode(priceCode)
    }

    fun setPriceCode(priceCode: Int) {
        price = when (priceCode) {
            REGULAR -> RegularPrice()
            NEW_RELEASE -> NewReleasePrice()
            CHILDRENS -> ChildrensPrice()
            else -> throw IllegalArgumentException("Incorrect Price Code")
        }
    }

    fun getCharge(daysRented: Int): Double {
        return price.getCharge(daysRented)
    }

    fun getFrequentRenterPoints(daysRented: Int): Int {
        return price.getFrequentRenterPoints(daysRented)
    }

    companion object {
        const val REGULAR = 0
        const val NEW_RELEASE = 1
        const val CHILDRENS = 2
    }
}